src/reader/reader_files.c

```diff
@@ -0,0 +1,59 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   reader_files.c                                     :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: alopez-g <marvin@42.fr>                    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2022/10/22 23:28:27 by alopez-g          #+#    #+#             */
+/*   Updated: 2022/10/22 23:28:37 by alopez-g         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "reader.h"
+
+static t_bool	is_quote_paired(char *str)
+{
+	char	quote_char;
+	size_t	i;
+
+	i = 1;
+	quote_char = *str;
+	while ((*(str + i) && *(str + i) != quote_char))
+		i++;
+	if (*(str + i) == quote_char)
+		return (i);
+	return (0);
+}
+
+static t_bool	is_quote_valid(char *line)
+{
+	size_t	i;
+
+	i = -1;
+	while (*(line + ++i))
+	{
+		if (reader_is_squote(*(line + i)) || reader_is_dquote(*(line + i)))
+		{
+			if (!is_quote_paired(line + i))
+				return (FALSE);
+			i++;
+		}
+	}
+	return (TRUE);
+}
+
+t_error_code	reader_get_tab_from_file(char ***ret)
```

Comment:
i have tested a file with an open '
it has a readline leak that im not sure about.
also there is no error printed like bash would.
![image](https://user-images.githubusercontent.com/49197152/197537849-fdd6a5d2-d7db-4ac3-ba86-0b2832ee2d2e.png)

but tbh, since we don't really need to implement file reading then maybe we can skip this issue